backend/onedrive/onedrive.go

```diff
@@ -142,6 +149,20 @@ func init() {
 			case "siteid":
 				fmt.Printf("Paste your Site ID here> ")
 				siteID = config.ReadLine()
+			case "url":
+				fmt.Println("Example: \"https://contoso.sharepoint.com/sites/mysite\" or \"mysite\"")
+				fmt.Printf("Paste your Site URL here> ")
+				siteURL := config.ReadLine()
+				re := regexp.MustCompile(`https://.*\.sharepoint.com/sites/(.*)`)
+				match := re.FindStringSubmatch(siteURL)
+				if len(match) == 2 {
+					relativePath = "/sites/" + match[1]
+				} else {
+					relativePath = "/sites/" + siteURL
+				}
+			case "path":
+				fmt.Printf("Enter server-relative URL here> ")
```

Comment:
Is this something users would know what it is? Maybe we need an example here?

Reply:
If you are looking for "what is server-relative URL", I don't think use would ever use it but will be very useful when we need it, like adding a Sharepoint group/team site. More or less this is the siteid equivalent but much more easy to find.

This option is the by-product of addressing the site by URL.

Here is an exmaple form MS Graph API Doc and some other cases
Assuming the user root site is `contoso.sharepoint.com`

| Path        | Description |
| ------------- |:-------------:|
| /teams/hr | The site associated with https://contoso.sharepoint.com/teams/hr |
| /teams/hr:/drive | Access the default **drive** for this site |
| /sites/mysite | The site associated with https://contoso.sharepoint.com/sites/mysite |

However, since we assume the path is pointing to a site, the second one, **pointing to a drive is not vaild**.

ref:
https://docs.microsoft.com/en-us/graph/api/site-getbypath?view=graph-rest-1.0
https://docs.microsoft.com/en-us/graph/api/resources/sharepoint?view=graph-rest-1.0#sharepoint-api-root-resources
https://stackoverflow.com/questions/45406451/how-can-i-get-the-siteid-of-the-current-site-with-microsoft-graph-api